src/lib/consensus/proof_of_stake.ml

```diff
@@ -754,33 +770,39 @@ module Make (Inputs : Inputs_intf) :
       in
       (last_data, curr_data, epoch_length)
 
-    let update_pair_checked (last_data, curr_data) epoch_length ~prev_epoch
-        ~next_epoch ~curr_slot ~prev_protocol_state_hash ~proposer_vrf_result
-        ~ledger_hash ~total_currency =
-      let open Snark_params.Tick.Let_syntax in
+    let _update_pair_checked (last_data, curr_data) epoch_length ~prev_epoch
+        ~next_epoch ~next_slot:_ ~curr_slot ~prev_protocol_state_hash
+        ~proposer_vrf_result ~new_ledger_hash ~new_total_currency =
+      let open Snark_params.Tick in
+      let open Let_syntax in
       let%bind last_data, curr_data, epoch_length =
-        let%bind epoch_changed =
-          Epoch.compare_var prev_epoch next_epoch >>| fun c -> c.less
+        let%bind epoch_increased =
+          let%bind c = Epoch.compare_var prev_epoch next_epoch in
+          let%map () = Boolean.Assert.is_true c.less_or_equal in
+          c.less
         in
-        let%map last_data = if_ epoch_changed ~then_:curr_data ~else_:last_data
+        let%map last_data =
+          if_ epoch_increased ~then_:curr_data ~else_:last_data
         and curr_data =
-          if_ epoch_changed
+          if_ epoch_increased
             ~then_:
-              { seed= Epoch_seed.(var_of_t (of_hash zero))
-              ; ledger= {hash= ledger_hash; total_currency}
+              { seed= Epoch_seed.(var_of_t initial)
+              ; ledger=
+                  {hash= new_ledger_hash; total_currency= new_total_currency}
               ; start_checkpoint= prev_protocol_state_hash
               ; lock_checkpoint= Coda_base.State_hash.(var_of_t (of_hash zero))
               ; length= Length.Unpacked.var_of_value Length.zero }
             ~else_:curr_data
         and epoch_length =
-          Length.increment_if_var epoch_length epoch_changed
+          Length.increment_if_var epoch_length epoch_increased
         in
         (last_data, curr_data, epoch_length)
       in
       let%map curr_seed, curr_lock_checkpoint =
         let%bind updated_curr_seed =
           Epoch_seed.update_var curr_data.seed proposer_vrf_result
         and in_seed_update_range =
+          (* TODO: Should this be next_slot? *)
```

Comment:
I think this is correct. Just double check your implementation against this issue and PR to make sure the way you wrote the code is compliant.
https://github.com/CodaProtocol/coda/issues/1022
https://github.com/CodaProtocol/coda/pull/1081